src/baldr/datetime.cc

```diff
@@ -162,19 +165,22 @@ std::string
 seconds_to_date(const uint64_t seconds, const date::time_zone* time_zone, bool tz_format) {
 
   std::string iso_date;
-  if (seconds == 0 || !time_zone) {
+  if (seconds == 0) {
     return iso_date;
   }
 
+  std::ostringstream iso_date_time;
+  const char* format = tz_format ? "%FT%R%z" : "%FT%R";
   std::chrono::seconds dur(seconds);
   std::chrono::time_point<std::chrono::system_clock> tp(dur);
-  const auto date = date::make_zoned(time_zone, tp);
 
-  std::ostringstream iso_date_time;
-  if (tz_format)
-    iso_date_time << date::format("%FT%R%z", date);
-  else
-    iso_date_time << date::format("%FT%R", date);
+  if (time_zone) {
+    const auto date = date::make_zoned(time_zone, tp);
+    iso_date_time << date::format(format, date);
+  } else {
+    iso_date_time << date::format(format, tp);
+  }
```

Comment:
same here, just give back the seconds without doing timezone stuff if there's none